### PR TITLE
rgw: usage dump_unsigned instead dump_int

### PR DESCRIPTION
**Pull request by joke-lee**
Signed-off-by: yuliyang <yuliyang@cmss.chinamobile.com>

### CODE REVIEW

**Comment by joke-lee** [comment]
hi @cbodley , would you mind take a review? thanks

**Comment by iliul** [comment]
@joke-lee Whether these two places also need to be convert to dump_unsigned ？
 
https://github.com/ceph/ceph/blob/912250b04ab1cadffc45c20524241bd3b20751b0/src/rgw/rgw_rest_s3.cc#L552-L553

and 

https://github.com/ceph/ceph/blob/912250b04ab1cadffc45c20524241bd3b20751b0/src/rgw/rgw_rest_s3.cc#L638-L639

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -536,10 +536,10 @@ static void dump_usage_categories_info(Formatter *formatter, const rgw_usage_log
...
     formatter->open_object_section("Entry");
     formatter->dump_string("Category", uiter->first);
-    formatter->dump_int("BytesSent", usage.bytes_sent);
-    formatter->dump_int("BytesReceived", usage.bytes_received);
-    formatter->dump_int("Ops", usage.ops);
-    formatter->dump_int("SuccessfulOps", usage.successful_ops);
+    formatter->dump_unsigned("BytesSent", usage.bytes_sent);
+    formatter->dump_unsigned("BytesReceived", usage.bytes_received);
+    formatter->dump_unsigned("Ops", usage.ops);
+    formatter->dump_unsigned("SuccessfulOps", usage.successful_ops);
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc]: it's probably better to use `encode_json()` in general, ie:
```c++
encode_json("SuccessfulOps", usage.successful_ops, formatter);
```
which has overloads that call `dump_int()` for signed types and `dump_unsigned()` for unsigned types

Comment (cbodley) [reply src/rgw/rgw_rest_s3.cc]: ping @joke-lee

**Comment by joke-lee** [comment]
close this, new pr in https://github.com/ceph/ceph/pull/28308

**Pull request closed without merging**